### DISCUSSION TITLE
Incorrect routing when the query parameter is in the configuration

### DISCUSSION THREAD

**Discussion by Burgyn** [ThreeMammals/Ocelot#2065] [Q&A] [answered]
Hi guys.

I discovered strange and different routing behavior between versions `22.0.1` and `23.0.0`.

## Expected Behavior

In version `22.0.1` the routing works well as expected.
We have the following configuration:

```json
{
    "DownstreamPathTemplate": "/{tenantId}/products?{everything}",
    "UpstreamPathTemplate": "/{tenantId}/products?{everything}",
    "UpstreamHttpMethod": [ "Get" ]
},
{
    "DownstreamPathTemplate": "/{tenantId}/products/{everything}",
    "UpstreamPathTemplate": "/{tenantId}/products/{everything}",
    "UpstreamHttpMethod": [ "Get" ]
},
```

When I call a request
```http
GET localhost/1/products/1
```

then the correct route `"/{tenantId}/products/{everything}"` is selected.

## Actual Behavior

In version `23.0.0` the `"DownstreamPathTemplate":"/{tenantId}/products?{everything}"` path is chosen for some reason, which is not correct.

## Specifications

  - Version: >23.0.0

---

Maybe this PR? [#748](https://github.com/ThreeMammals/Ocelot/pull/1911)

Any idea for a quick fix on my part?
Thank you very much.

**Comment by raman-m** [comment, marked as answer]
Eliminate the initial redundant route as it is inherently included in the subsequent Catch All route.

It should be one route only
```json
{
    "DownstreamPathTemplate": "/{tenantId}/products/{everything}", // all query strings are included
    "UpstreamPathTemplate": "/{tenantId}/products/{everything}",
    "UpstreamHttpMethod": [ "Get" ]
},
```

Also, note that if no [Priority](https://ocelot.readthedocs.io/en/latest/features/routing.html#priority) is defined, then the first matching route prevails! This explains the odd behavior observed.

**Reply by raman-m** [reply]
### Reported once again
- #2132

**Reply by raman-m** [reply]
@Burgyn Although I haven't been able to reproduce this bug yet, we aim to address it in the [September'24](https://github.com/ThreeMammals/Ocelot/milestone/7) or, at the latest, in the [October'24](https://github.com/ThreeMammals/Ocelot/milestone/9) while working on issue #2132 which is prioritized now. I am also open to any contributions you may have.

**Reply by raman-m** [reply]
🤣 Reported once again: #2169

**Comment by raman-m** [comment]
@Burgyn Hello Miňo!
The bug will be fixed by &rarr;
- #2225
